frontend/src/components/ConcentrationDropdown.tsx

```diff
@@ -0,0 +1,139 @@
+import { FormControl, FormHelperText, TextField } from "@material-ui/core";
+import { Autocomplete } from "@material-ui/lab";
+import { CSSProperties } from "@material-ui/styles";
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import styled from "styled-components";
+import { Concentration, Major } from "../../../common/types";
+import {
+  safelyGetActivePlanMajorObjectFromState,
+  getUserMajorFromState,
+  getUserConcentrationFromState,
+  safelyGetActivePlanConcentrationFromState,
+} from "../state";
+import { setUserConcentrationAction } from "../state/actions/userActions";
+import { setActivePlanConcentrationAction } from "../state/actions/userPlansActions";
+import { AppState } from "../state/reducers/state";
+
+interface SaveInParentConcentrationDropdownProps {
+  readonly major: Major | undefined;
+  readonly concentration: string | null;
+  readonly setConcentration: (concentration: string | null) => void;
+  readonly setError?: (error: boolean) => void; // To tell the parent that there is an error with the input (no major selected when there should be)
+  readonly style?: CSSProperties;
+  readonly useLabel?: boolean;
+  readonly showError?: boolean;
+}
+
+interface SaveOnChangeConcentrationDropdownProps {
+  readonly isUserLevel: boolean;
```

Comment:
Would be nice to have a comment or something here saying like "otherwise, it saves the concentration to the active plan"